service/policy/db/key_access_server_registry.go

```diff
@@ -296,3 +297,305 @@ func (c PolicyDBClient) ListKeyAccessServerGrants(ctx context.Context, r *kasreg
 		},
 	}, nil
 }
+
+/*
+* Key Access Server Keys
+ */
+func (c PolicyDBClient) CreateKey(ctx context.Context, r *kasregistry.CreateKeyRequest) (*kasregistry.CreateKeyResponse, error) {
+	keyID := r.GetKeyId()
+	algo := int32(r.GetKeyAlgorithm())
+	mode := int32(r.GetKeyMode())
+	privateCtx := r.GetPrivateKeyCtx()
+	pubCtx := r.GetPublicKeyCtx()
+	providerConfigID := r.GetProviderConfigId()
+	keyStatus := int32(policy.KeyStatus_KEY_STATUS_ACTIVE)
+
+	kasID := r.GetKasId()
+	_, err := c.GetKeyAccessServer(ctx, &kasregistry.GetKeyAccessServerRequest_KasId{
+		KasId: kasID,
+	})
+	if err != nil {
+		return nil, db.StatusifyError(err, db.ErrTextGetRetrievalFailed, kasID)
+	}
+
+	// Only allow one active key for an algo per KAS.
+	activeKeyExists, err := c.Queries.CheckIfKeyExists(ctx, CheckIfKeyExistsParams{
+		KeyAccessServerID: kasID,
+		KeyStatus:         keyStatus,
+		KeyAlgorithm:      algo,
+	})
+	if err != nil {
+		return nil, db.WrapIfKnownInvalidQueryErr(err)
+	} else if activeKeyExists {
+		return nil, fmt.Errorf("cannot create a new key when an active key already exists with algorithm %s", r.GetKeyAlgorithm().String())
+	}
+
+	// PrivateCTX should be > 0 len
+	if len(privateCtx) < 1 {
+		return nil, fmt.Errorf("private key context is required")
+	}
+
+	// Verify that publicKey is in the request if key is mode local
+	if mode == int32(policy.KeyMode_KEY_MODE_LOCAL) && len(pubCtx) < 1 {
```

Comment:
We will always want a public key. Otherwise the platform will need to grab it from aws kms when we map the key to a value or namespace.